js/spreadsheet.js

```diff
@@ -242,6 +242,23 @@ var spreadsheet = function(options) {
     },
     afterSelectionEnd: function(r, c, r2, c2) {
       s = [r, c, r2, c2];
+    },
+    beforeKeyDown: function (event) {
+      event = event || window.event;
+
+      if ((event.ctrlKey || event.metaKey) && event.keyCode === 65 ) {
```

Comment:
So, this solution looks like when `Ctrl + A` or `Cmd + A` are pressed, all cells are selected.

What happens if the user is currently editing a cell? If that happens, I think users would be expecting for all contents within the cell to be selected like usual spreadsheet behaviour. Does this happen?

Reply:
@tonytlwu 

> Does this happen?

No, but I believe I have a solution for this scenario.
We should add an additional variable called `isEditing` with the default value `false`.
In the `handsontable` library, we had a method `beginEditing()` which is calling after a user will begin editing a cell. In this method, we will change `isEditing` variable to `true` and after the user will exit the editing mode. Which will call the `finishEditing()` method, there we will change `isEditing` to the `false` once again.

And alter this method like this:
```
    beforeKeyDown: function (event) {
     if (isEditing){
        return;
      }
     
      event = event || window.event;

      if ((event.ctrlKey || event.metaKey) && event.keyCode === 65 ) {
        event.stopImmediatePropagation();
        
        var cols = getColumns().filter(function(column) {
          return column;
        }).length;
        var colEnd = Math.max(cols - 1, 0);
        var rowEnd = getData().length;

        hot.deselectCell();
        hot.selectCells([[0, 0, rowEnd, colEnd]], false, false);
        return false;
      }
    }
```

Will it be a good solution?

Reply:
@tonytlwu 
Ignore the previous comment I've found a simpler way.

In the hot object, we have a `getActiveEditor()` method in which we can access the `_opened`
option and it indicates are a user edits a cell or not. It will return `true` or `false`.

So the solution is:
```
    beforeKeyDown: function (event) {
      if (hot.getActiveEditor()._opened){
        return;
      }

      event = event || window.event;

      if ((event.ctrlKey || event.metaKey) && event.keyCode === 65 ) {
        event.stopImmediatePropagation();
        
        var cols = getColumns().filter(function(column) {
          return column;
        }).length;
        var colEnd = Math.max(cols - 1, 0);
        var rowEnd = getData().length;

        hot.deselectCell();
        hot.selectCells([[0, 0, rowEnd, colEnd]], false, false);
        return false;
      }
    }
```

In this case, the user will be able to select a text within the cell while he edits it.

Reply:
Yes, that looks like a good recommendation 👍